### PR TITLE
BCE - 14041 add log analytics

### PR DESCRIPTION
**Pull request by mayshavit**

### CODE REVIEW

File: src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private val LOG = logger<AnalyticsService>()
+
+    private lateinit var fullScanData: FullScanData
+    private var fullScanNumber = 0
+
+    fun fullScanButtonWasPressed() {
+        fullScanNumber += 1
+        LOG.info("Prisma Plugin Analytics - scan #${fullScanNumber} - full scan button was pressed")
+        fullScanData = FullScanData(fullScanNumber)
+        fullScanData.fullScanButtonPressedTime = Date()
```
Comment (saarschweid) [inline src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt:20]: I would put that on the first line, currently not much difference but if the logic gets more complicated you want the time to reflect the click

File: src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        val minutes = TimeUnit.MILLISECONDS.toMinutes(endTime.time - startTime.time)
+        val seconds = TimeUnit.MILLISECONDS.toSeconds(endTime.time - startTime.time) - (minutes * 60)
+        val secondsString = if (seconds < 10) {
+            "0${seconds}"
+        } else "${seconds}"
+        return "${minutes}:${secondsString}"
+    }
+
+    data class FullScanData(val scanNumber: Int) {
+        lateinit var fullScanButtonPressedTime: Date
```
Comment (saarschweid) [inline src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt:112]: this object collects data per scan type right? not the entire run.
Maybe we want to keep one object (as `object`) and update its fields for the entire metadata.
I can use it later in the top bar implementation

Comment (saarschweid) [inline src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt:112]: dont we collect the results (failed / succeeded / total)?

**Review by saarschweid — APPROVED** [review]

File: src/main/kotlin/com/bridgecrew/ui/actions/CheckovScanAction.kt [hunk 3]
```diff
@@ -11,6 +12,7 @@ class CheckovScanAction : AnAction(AllIcons.Actions.Execute), DumbAware {
 
     override fun actionPerformed(actionEvent: AnActionEvent) {
         val project = actionEvent.project
-        project?.service<CheckovScanService>()?.scanProject(project)
+        project!!.service<AnalyticsService>().fullScanButtonWasPressed()
+        project.service<CheckovScanService>().scanProject(project)
```
[the remark this answers is not in the document]
Comment (mayshavit) [reply src/main/kotlin/com/bridgecrew/ui/actions/CheckovScanAction.kt:16]: Yes, here is specific for the UI, we have another analytics for the full scan itself - fullScanStarted

File: src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt (hunk 2 above)
Comment (mayshavit) [reply src/main/kotlin/com/bridgecrew/analytics/AnalyticsService.kt:113]: currently not but it can definitely be added later

**Pull request merged**